test/lib/ansible_test/_util/controller/sanity/validate-modules/validate_modules/schema.py

```diff
@@ -47,6 +47,8 @@ def isodate(v, error_code=None):
 
 
 COLLECTION_NAME_RE = re.compile(r'^([^.]+(\.[^.]+)+)$')
+FULLY_QUALIFIED_COLLECTION_RESOURCE_RE = re.compile(r'^([^.]+\.[^.]+(\.[^.]+)+)$')
+URL_RE = re.compile(r'''^(?:(?:https?:)?//[^/?#]*|)[^?#]*(?:\?([^#]*))?(?:#(.*))?$''')
```

Comment:
I am slightly wary of how complicated URLs can be and whether this might end up with false positives in rare cases. Is there a library or something else we can use to validate a URL instead?

Reply:
I guess https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse would be the best choice. Though there's not much it won't accept (even if we restrict the schema to one of `'http'`, `'https'` and `None`).

Reply:
ok, just something for consideration, I trust your/the community's judgement on it

Reply:
I switched to urlparse in 937d40f26d6fca652165f99d8baacc0a185d1007.